packages/workspace/src/core/hasher/hasher.ts

```diff
@@ -148,12 +162,149 @@ export class Hasher {
     return this.runtimeInputs;
   }
 
+  private tsconfigDepsHash(
+    sortedTransitiveDeps: Array<[string, ProjectGraphNode]>,
+    workspaceJson: any
+  ): Array<{
+    hash: string;
+    record: string;
+  }> {
+    const npmScope = `@${this.nxJson.npmScope}/`;
+    const tsConfig = readTsConfig();
+
+    /**
+     * Parse out the paths from tsconfig so they are not accounted for
+     * in hashing tsconfig configurations since they are hashed separately
+     */
+    const { paths, ...compilerOptions } = tsConfig.compilerOptions;
+    tsConfig.compilerOptions = compilerOptions;
+
+    const hashedTsConfig: {
+      hash: string;
+      record: string;
+    } = {
+      hash: this.hashing.hashObject(sortObject(tsConfig)),
+      record: 'tsconfig.base.json',
+    };
+
+    const hashedTsConfigPaths: {
+      hash: string;
+      record: string;
+    } = {
+      hash: this.hashing.hashArray(
+        sortedTransitiveDeps.map(([name]) => {
+          const project = workspaceJson.projects[name];
+          const path = project.root.replace(
+            project.projectType === 'application' ? 'apps/' : 'libs/',
+            ''
+          );
+          return paths[`${npmScope}${path}`];
+        })
+      ),
+      record: 'tsconfig.base.json[compilerOptions][paths]',
```

Comment:
Wasn't sure what to use as the "source" property for this and the `hashedTsConfig` above so I just chose a unique value for this specific hash. It can be changed to whatever.